### PR TITLE
Use tacc/core-styles

### PR DESCRIPTION
**Pull request by nathanfranklin**
## Overview: ##

Add core styles.  

Input from Wes from meeting https://tacc-main.atlassian.net/wiki/spaces/DES/pages/43384853/2024-01-25+Meeting+notes

## PR Status: ##

* [X] Ready.

### CODE REVIEW

File: react/src/core-components/TextCopyField/TextCopyField.module.css [hunk 1]
```diff
@@ -7,7 +7,7 @@
   transition: color var(--transition-duration),
     background-color var(--transition-duration);
 
-  composes: c-button--secondary from '@tacc/core-styles/src/lib/_imports/components/c-button.css';
+  composes: c-button--secondary from '@tacc/core-styles/dist/components/c-button.css';
```
Comment (nathanfranklin) [inline react/src/core-components/TextCopyField/TextCopyField.module.css:10]: @ Wes  , @tjgrafft , @sophia-massie  This was causing the errors we were seeing; so should make its way back to TUP and/or the future core-components package:

```
GET http://localhost:4200/tools/x-truncate.css net::ERR_ABORTED 404 (Not Found)
GET http://localhost:4200/objects/o-section.selectors.css net::ERR_ABORTED 404 (Not Found)
GET http://localhost:4200/tools/x-link.css net::ERR_ABORTED 404 (Not Found)
GET http://localhost:4200/c-button.selectors.css net::ERR_ABORTED 404 (Not Found)
```

**Review by tjgrafft — APPROVED** [review]
LGTM! Tested locally. Console errors/warnings are gone.

**Review by shayanaijaz — APPROVED** [review]
LGTM

File: react/package.json [hunk 2]
```diff
@@ -24,7 +24,7 @@
         "@babel/preset-typescript": "^7.21.0",
         "@changey/react-leaflet-markercluster": "^4.0.0-rc1",
         "@reduxjs/toolkit": "^1.8.4",
-        "@tacc/core-styles": "^2.22.6",
+        "@tacc/core-styles": "^2.22.3",
```
Comment (wesleyboar) [inline react/package.json:27]: Why the downgrade? What did I add after 2.22.3 that smelled so bad? 😄 Latest release currently is 2.23.1.

File: react/src/core-components/TextCopyField/TextCopyField.module.css (hunk 1 above)
Comment (wesleyboar) [reply react/src/core-components/TextCopyField/TextCopyField.module.css:10]: Thanks, yes. Bug in TUP. _The `/dist/` is preferred unless one need do something advanced._

File: react/src/index.css [hunk 3]
```diff
@@ -1,70 +1,9 @@
...
-  font-size: 16px;
-  line-height: 24px;
-  font-weight: 400;
+@layer foundation, base, project;
 
-  color-scheme: light dark;
-  color: rgba(255, 255, 255, 0.87);
-  background-color: #242424;
+@import url('https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/css/bootstrap.min.css')
+layer(foundation);
```
Comment (wesleyboar) [inline react/src/index.css:4]: I am unsure if ` layer(…)` works if on a second line. Have you tested this?

```suggestion
@import url('https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/css/bootstrap.min.css') layer(foundation);
```

File: react/src/index.css [hunk 4]
```diff
@@ -1,70 +1,9 @@
...
-}
-button:hover {
-  border-color: #646cff;
-}
-button:focus,
-button:focus-visible {
-  outline: 4px auto -webkit-focus-ring-color;
-}
-
-@media (prefers-color-scheme: light) {
```
Comment (wesleyboar) [inline react/src/index.css:59]: **Note.** Core-Styles does not offer `@media (prefers-color-scheme: …)` yet. _I want it. M.S. wants it. Say you do to, then maybe I get to work on it._

File: react/src/pages/MainMenu/MainMenu.tsx [hunk 5]
```diff
@@ -32,12 +33,20 @@ function MainMenu() {
...
+          <tr>
+            <th>Projects</th>
+          </tr>
+        </thead>
+        <tbody>
+          <tr>
+            <td>You have {data?.length} projects.</td>
+          </tr>
+        </tbody>
       </table>
```
Comment (wesleyboar) [inline react/src/pages/MainMenu/MainMenu.tsx:50]: **Yes, great!** Accurate markup for tables is important to apply Core-Styles to them.

**Review by wesleyboar — APPROVED** [review]

**Review by wesleyboar — CHANGES_REQUESTED** [review]
That one new line before ` layer(foundation)` worries me enough to reconsider merge. Please check https://github.com/TACC-Cloud/hazmapper/pull/194#discussion_r1480653282. I don't want y'all to have to fight Bootstrap because of a typo.

File: react/src/index.css (hunk 4 above)
Comment (nathanfranklin) [reply react/src/index.css:59]: I think this was just Ian having placeholder CSS to get us up an doing so probably best not to consider it.

File: react/package.json (hunk 2 above)
Comment (nathanfranklin) [reply react/package.json:27]: I'll bump to latest release. i think we were just matching TUP for no reason.

**Review by wesleyboar — APPROVED** [review]

File: react/package.json (hunk 2 above)
Comment (wesleyboar) [reply react/package.json:27]: Okay. Safety. Makes sense. Core-Styles follows [SemVer](https://semver.org/), so you may trust `X.`**`Y`**`.Z` and `X.Y.`**`Z`** to not break anything. _Example: I delay `2.24.0` release until I fix bugs on redesigned tables._

**Pull request merged**